app/src/main/java/com/infomaniak/drive/ui/addFiles/CreatePrivateFolderFragment.kt

```diff
@@ -44,14 +45,15 @@ class CreatePrivateFolderFragment : CreateFolderFragment() {
                 file?.let {
                     saveNewFolder(file)
                     requireActivity().showSnackbar(R.string.createPrivateFolderSucces, anchorView = requireActivity().mainFab)
-                    safeNavigate(
-                        if (redirectToShareDetails) CreatePrivateFolderFragmentDirections.actionCreatePrivateFolderFragmentToFileShareDetailsFragment(
-                            file = file, ignoreCreateFolderStack = true
+                    if (redirectToShareDetails) {
+                        safeNavigate(
+                            CreatePrivateFolderFragmentDirections.actionCreatePrivateFolderFragmentToFileShareDetailsFragment(
+                                file = file, ignoreCreateFolderStack = true
+                            )
                         )
-                        else CreatePrivateFolderFragmentDirections.actionCreatePrivateFolderFragmentToFileListFragment(
-                            folderID = file.id, folderName = file.name, ignoreCreateFolderStack = true
-                        )
-                    )
+                    } else {
+                        findNavController().popBackStack(R.id.newFolderFragment, true)
```

Comment:
With your `findNavController().popBackStack(R.id.newFolderFragment, true)`
The back button does not work anymore

Reply:
Fixed.